refbox/src/tournament_manager/mod.rs

```diff
@@ -1715,6 +1775,174 @@ impl TournamentManager {
+            GamePeriod::SuddenDeath => self.clock_state.clock_time(now).unwrap(),
+            GamePeriod::SecondHalf | GamePeriod::OvertimeSecondHalf => Duration::from_millis(10),
+            _ => unreachable!(),
+        };
+
+        self.clock_state = ClockState::Stopped {
+            clock_time: Duration::from_millis(10),
+        };
+
+        self.time_pause_confirmation = Some(ConfirmPause {
+            pause_began: pause_inst,
+            duration_of_pause: dur_pause,
+            clock_time: clock_at_pause,
+        });
+
+        Ok(())
+    }
+
+    /// Scores must be accurately set before calling this
+    pub fn end_confirm_pause(&mut self, now: Instant) -> Result<()> {
+        info!(
+            "Ending Pause, Pause Duration: {:?}",
+            self.time_pause_confirmation
+        );
+        info!("Start Period: {}", self.current_period);
+        if let Some(confirm_pause) = &self.time_pause_confirmation {
+            let scores = self.scores;
+            info!("Scores: {:?}", scores);
+            self.current_period = match self.current_period {
+                GamePeriod::SecondHalf => {
+                    if scores.are_not_equal() {
+                        GamePeriod::BetweenGames
+                    } else if self.config.overtime_allowed {
+                        GamePeriod::PreOvertime
+                    } else if self.config.sudden_death_allowed {
+                        GamePeriod::PreSuddenDeath
+                    } else {
+                        GamePeriod::BetweenGames
+                    }
+                }
+                GamePeriod::OvertimeSecondHalf => {
+                    if scores.are_not_equal() {
+                        GamePeriod::BetweenGames
+                    } else if self.config.sudden_death_allowed {
+                        GamePeriod::PreSuddenDeath
+                    } else {
+                        GamePeriod::BetweenGames
+                    }
+                }
+                GamePeriod::SuddenDeath => {
+                    if !scores.are_not_equal() {
+                        GamePeriod::SuddenDeath
+                    } else {
+                        GamePeriod::BetweenGames
+                    }
+                }
+                _ => {
+                    unreachable!()
+                }
+            };
+
+            info!("Current Period: {}", self.current_period);
+
+            let pause_duration = now.duration_since(confirm_pause.pause_began);
+            let time_into_sd = confirm_pause.clock_time;
+
+            info!("Pause Duration 2: {:?}", pause_duration);
+
+            let next_period_remaining_duration = if self.current_period == GamePeriod::BetweenGames
+            {
+                let mut time_to_next_game =
+                    self.calc_time_to_next_game(now, confirm_pause.pause_began);
+                time_to_next_game -= pause_duration;
+                time_to_next_game
```

Comment:
```suggestion
                self.calc_time_to_next_game(now, confirm_pause.pause_began).saturating_sub(pause_duration)
```

the `-` here could panic if `pause_duration` is too large and would cause the result to be negative